src/storage/src/hummock/sstable/builder.rs

```diff
@@ -193,32 +193,35 @@ impl<W: SstableWriter, F: FilterBuilder> SstableBuilder<W, F> {
     }
 
     /// Add kv pair to sstable.
-    pub fn add_monotonic_delete(&mut self, event: MonotonicDeleteEvent) {
+    pub fn add_monotonic_delete(&mut self, mut event: MonotonicDeleteEvent) {
         let table_id = event.event_key.left_user_key.table_id.table_id();
         if self.last_table_id.is_none() || self.last_table_id.unwrap() != table_id {
             self.table_ids.insert(table_id);
         }
-        if event.new_epoch == MAX_EPOCH
+        if is_max_epoch(event.new_epoch)
             && self.monotonic_deletes.last().map_or(true, |last| {
-                last.new_epoch == MAX_EPOCH
+                is_max_epoch(last.new_epoch)
                     && last.event_key.left_user_key.table_id
                         == event.event_key.left_user_key.table_id
             })
         {
             // This range would never delete any key so we can merge it with last range.
             return;
         }
-        if event.new_epoch != MAX_EPOCH {
+        if !is_max_epoch(event.new_epoch) {
             self.epoch_set.insert(event.new_epoch);
         }
+        if is_compatibility_epoch(event.new_epoch) {
```

Comment:
question:
if we check epoch by `is_max_epoch`, shall we need to rewrite `event.new_poch` to HummockEpoch::MAX ? 

Besides the max value, are there any other comparisons?

Reply:
I think it is dangerous to mix two kind of MAX value in data.